File: backend/app/schemas/bookstore.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class BookCreate(BaseModel):
    title: str
    author: str | None = None
    genre: str | None = None
    isbn: str | None = None
    description: str | None = None
    quantity: int = 0
    price: Decimal | None = None


class BookUpdate(BaseModel):
    title: str | None = None
    author: str | None = None
    genre: str | None = None
    isbn: str | None = None
    description: str | None = None
    quantity: int | None = None
    price: Decimal | None = None


class BookResponse(BaseModel):
    id: int
    agent_id: int
    title: str
    author: str | None
    genre: str | None
    isbn: str | None
    description: str | None
    quantity: int
    price: Decimal | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class InventoryLogResponse(BaseModel):
    id: int
    book_id: int
    agent_id: int
    change_type: str
    quantity_change: int
    note: str | None
    performed_by: int | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/api/bookstore.ts
import api from './client'

export interface Book {
  id: number
  agent_id: number
  title: string
  author: string | null
  genre: string | null
  isbn: string | null
  description: string | null
  quantity: number
  price: number | null
  created_at: string
  updated_at: string
}

export interface InventoryLogEntry {
  id: number
  book_id: number
  agent_id: number
  change_type: string
  quantity_change: number
  note: string | null
  performed_by: number | null
  created_at: string
}

export async function getBooks(agentId: number): Promise<Book[]> {
  const { data } = await api.get(`/agents/${agentId}/bookstore/books`)
  return data
}

export async function createBook(agentId: number, body: Partial<Book>): Promise<Book> {
  const { data } = await api.post(`/agents/${agentId}/bookstore/books`, body)
  return data
}

export async function updateBook(agentId: number, bookId: number, body: Partial<Book>): Promise<Book> {
  const { data } = await api.put(`/agents/${agentId}/bookstore/books/${bookId}`, body)
  return data
}

export async function getInventoryLog(agentId: number): Promise<InventoryLogEntry[]> {
  const { data } = await api.get(`/agents/${agentId}/bookstore/inventory-log`)
  return data
}
